modules/ROOT/pages/dataweave-type-reuse-xml-schema.adoc

```diff
@@ -0,0 +1,430 @@
+
+
+                <xs:element name="date" type="xs:date"/>
+                <xs:element name="dateTime" type="xs:dateTime"/>
+                <xs:element name="time" type="xs:time"/>
+
+
+                <xs:element name="boolean" type="xs:boolean"/>
+
+
+                <xs:element name="hexBinary" type="xs:hexBinary"/>
+                <xs:element name="base64Binary" type="xs:base64Binary"/>
+                <xs:element name="decimal" type="xs:decimal"/>
+
+
+
+
+
+
+
+
+            </xs:all>
+        </xs:complexType>
+    </xs:element>
+</xs:schema>
+----
+
+.Generated DataWeave Module:
+[source,dataweave,linenums]
+----
+%dw 2.0
+
+
+type RootElementDefinition = {| Root: { string: String, normalizedString: String, integer: Number, negativeInteger: Number, int: Number, unsignedInt: Number, unsignedLong: Number, unsignedShort: Number, float: Number, double: Number, date: LocalDateTime, dateTime: DateTime, time: Time, boolean: Boolean, hexBinary: Binary, base64Binary: Binary, decimal: Number } |}
+
+
+type Root = RootElementDefinition
+----
+
+=== Root Elements Declaration
+
+Each root element (`<xs:element>`) declared inside your XSD file generates the following:
+
+* A DataWeave `type` directive with the element name followed by the `ElementDefinition` extension.
+* A `type` directive named `Root` as a `Union` type that consists of all the root elements in the XSD.
+
+The following example shows how to declare root elements:
+
+.XSD:
+[source,XML]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+
+
+    <xs:element name="Root">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element type="xs:string" name="color" default="Red"/>
+                <xs:element type="xs:integer" name="age" default="12"/>
+            </xs:sequence>
+            <xs:attribute type="xs:string" name="type" default="ALBA"/>
+        </xs:complexType>
+    </xs:element>
+
+
+    <xs:element name="AnotherRoot">
+        <xs:complexType>
+            <xs:sequence>
+                <xs:element type="xs:string" name="name" default="Red"/>
+            </xs:sequence>
+            <xs:attribute type="xs:string" name="type"/>
+        </xs:complexType>
+    </xs:element>
+</xs:schema>
+----
+
+.Generated DataWeave Module:
+[source,dataweave,linenums]
+----
+%dw 2.0
+
+
+type AnotherRootElementDefinition = {| AnotherRoot @("type"?: String): {- name: String -} |}
+
+
+type RootElementDefinition = {| Root @("type"?: String): {- color: String, age: Number -} |}
+
+
+type Root = AnotherRootElementDefinition | RootElementDefinition
+----
+
+=== Complex Types
+
+Named complex types declared as top-level elements on the schema which are referenced inside your schema or are available for reuse on other structures generate their own type directive. These type directives are named with the `complextype` name followed by the `Definition` extension.
```

Comment:
Long drink of water...

Named complex types (a bunch of modifiers) generate their own type directive.

What about:

Named complex types generate their own type directive when:
* blah
* dee
* dah